src/stores/comments.js

```diff
@@ -1,34 +1,115 @@
-import { collection, getDoc, getDocs } from 'firebase/firestore'
+import { collection, doc, getDoc, getDocs, query, setDoc, runTransaction, where, Timestamp, deleteDoc } from 'firebase/firestore'
 import { defineStore } from 'pinia'
-import { LocalStorage } from 'quasar'
+import { comment } from 'postcss'
 import { db } from 'src/firebase'
+import { useUserStore } from 'src/stores'
 
 export const useCommentStore = defineStore('comments', {
   state: () => ({
     _comments: [],
+    _entries: [],
     _isLoading: false
   }),
 
   getters: {
-    getComments: (state) => LocalStorage.getItem('comments') || state._comments,
+    getComments: (state) => state._comments,
+    getEntries: (state) => state._entries,
     isLoading: (state) => state._isLoading
   },
 
   actions: {
-    async fetchComments(entryId) {
-      console.log(entryId)
+    async fetchComments(slug) {
+      this._isLoading = true
+      const querySnapshot = await getDocs(query(collection(db, 'entries'), where('slug', '==', slug)))
+      const entry = querySnapshot.docs.map((doc) => ({ id: doc.id, ...doc.data() }))[0]
+
+      const c = query(collection(db, 'entries', entry.id, 'comments'))
+      const snap = await getDocs(c)
+      const comments = snap.docs.map((doc) => ({ id: doc.id, ...doc.data() }))
+
+      for (const comment of comments) {
+        comment.author = await getDoc(comment.author).then((doc) => doc.data())
+      }
+      this._isLoading = false
+
+      this._comments = comments
     },
 
-    async addComment(comment) {
-      console.log(comment)
+    async addComment(comment, entry) {
+      const userStore = useUserStore()
+      await userStore.loadBrowserId()
+
+      comment.author = !userStore.isAuthenticated ? userStore.getUserIp : userStore.getUserRef
+      comment.id = comment.author.id + Date.now()
+      comment.created = Timestamp.fromDate(new Date())
+      comment.showEdit = false
```

Comment:
You shouldn't save id inside the document to avoid duplications.
And `showEdit` should be only int the frontend. It is not needed in the Database.

It is just FYI, I am fixing all of this